docs/guide/SecurityBC.md

```diff
@@ -273,3 +273,302 @@ Here is the response that comes back:
+
+
+Some possible debugging actions that have been found useful in different circumstances:
+
+
+* Make a match much more permissive (replacing the entire piece beginning with `<.*>` and then having the minimum currently unique suffix is good).
+
+* Double-check that each internal URL is the proper internal URL by checking it is accessible inside the cluster with curl.
+
+* Look at the identity provider (IdP) logs.
+
+* Make sure that the domains and ports for introspection and the external token endpoint are identical. Keycloak at least does not like it if they are not.
+
+* Point the Oathkeeper rule at https://httpbin.org/, usually the `/anything` path prefix which will reflect back everything it gets, making it easy to see what the service will see.
+
+
+### Components required in addition to a Helm chart
+
+The following pieces will be needed in addition to the Helm chart:
+
+* a JWKS secret
+* annotated Helm values
+
+
+#### JWKS secret
+
+
+A secret should be created with key `mutator.id_token.jwks.json` and the value of a JWKS suitable for use with Oathkeeper. An initial one can be generated as described in [Configure and Deploy | ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/configure-deploy#cryptographic-keys). It will contain both public and private keys.
+
+
+##### Operating with the JWKS secret
+
+
+To rotate the secret, apply the following procedure:
+
+0. Note the time.
+1. Add a public and private key pair to the beginning of the array in the JWKS (make sure all public JWKs have a specified unique `kid`) in the secret. All keys other than the new keys and the previous first public and private keys can be removed. This is because Oathkeeper always signs with the first key.
+2. Wait until all requests that Oathkeeper might have received and authorized would have had their JWT handled by the backend service. The main delay here is the time for the secret update to propagate, comprising the delay of the secret manager to the updated secret and the delay of the secret to the updated volume, which is probably at most a minute or two, so wait that long after the time in step 0.
+3. If the old secret is being removed (this is only necessary if a breach is suspected, otherwise step 1 is sufficient for periodic key rotation), remove it now.
+
+#### Annotated Helm values
+
+
+Several places will need to be changed to the rest-of-deployment-specific URLs or other values. Those places are described in the comments in the example below, as well as other commentary.
+
+How to setup the Proxy ingress is undecided at the time, as it will need to change when the solution is added to the IaC 3.xxx so that area of the config is still unspecified. This leaves the ingress out by default. Changing `ingress.proxy.enabled` to `true` will enable the proxy ingress. See the linked pages at the beginning for options  available for the built-in ingress configuration.

```

Comment:
```suggestion
How to setup the Proxy ingress is undecided at the time, as it will need to change when the solution is added to the IaC 3.xxx so that area of the config is still unspecified. This leaves the ingress out by default. Changing `ingress.proxy.enabled` to `true` will enable the proxy ingress. See the linked pages at the beginning for options available for the built-in ingress configuration.

```
Apart from the above minor edit, it's all good now.